server/spawn-vehicle.lua

```diff
@@ -13,19 +15,49 @@ end
 ---@param garageName string
 ---@param accessPointIndex integer
 ---@return number? netId
-lib.callback.register('qbx_garages:server:spawnVehicle', function (source, vehicleId, garageName, accessPointIndex)
+lib.callback.register('qbx_garages:server:spawnVehicle', function(source, vehicleId, garageName, accessPointIndex)
+    -- Prepare the hook payload with the possibility to override configurations
+    local hookPayload = {
+        source = source,
+        vehicleId = vehicleId,
+        garageName = garageName,
+        accessPointIndex = accessPointIndex,
+        distanceCheck = Config.distanceCheck, -- Default value from config
+        doorsLocked = Config.doorsLocked, -- Default value from config
+        cancel = false,
+        veh = nil,
+        netId = nil,
+        vehiclekeys = true,
+        Notify = { -- For custom notifications
+            description = nil,
+            type = 'inform'
+        },
+    }
+
+    -- Trigger the hook and allow it to modify the payload (optional)
+    if triggerEventHooks('garages:spawnVehicle', hookPayload) == false then
+        lib.print.warn("Vehicle spawn was canceled by a hook.")
+        if hookPayload.Notify.description then
+            return exports.qbx_core:Notify(source, hookPayload.Notify.description, hookPayload.Notify.type)
+        end
```

Comment:
> Notify.description will always be nil. This should be deleted altogether as other resources registering hooks can add notifications in the hook callback function.

OK, so it would be better to do away with it and manage it that way?

```lua
exports.qbx_garages:registerHook('garages:spawnedVehicle', function(payload)
    exports.qbx_core:Notify(source, 'Enjoy your trip.', 'inform')
    return true -- Continue with the spawn
end)
```

Reply:
Yes exactly.